Missing support for .mjs files

We'd like to move to newer nodejs 18 and 20 which is supported by AWS now. Our setup:

package.json :
```json
{
  "name": "fk-demo-connector",
  "version": "1.0.0",
  "description": "",
  "type": "module",
  "main": "index.mjs",
}
```

index.mjs :
```javascript
console.log('Loading function');

export const handler = async (event, context) => {
    console.log('Received event:', JSON.stringify(event, null, 2));
   
    return {};  // Echo back the first key value
    // throw new Error('Something went wrong');
};
```

Running node-lambda:
```
> node-lambda run --contextFile ../context.json -f ../.env -j event-init.json

node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/vlad/workspace/lambda/fk-demo-connector/index.js' imported from /usr/local/lib/node_modules/node-lambda/lib/main.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:327:11)
    at moduleResolve (node:internal/modules/esm/resolve:946:10)
    at defaultResolve (node:internal/modules/esm/resolve:1132:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:524:22)
    at importModuleDynamically (node:internal/modules/cjs/loader:1188:29)
    at importModuleDynamicallyWrapper (node:internal/vm/module:429:21) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.18.0
```

Suggestions:
* Allow to parameterize extension index.js -> index.mjs
* Take index.mjs from package.json main key

--- a/index_mjs.mjs
+++ b/index_mjs.mjs
@@ -1,0 +1,9 @@
+// For development/testing purposes
+export const handler = (event, context, callback) => {
+  console.log('Running index.handler (mjs)')
+  console.log('==================================')
+  console.log('event', event)
+  console.log('==================================')
+  console.log('Stopping index.handler (mjs)')
+  callback(null)
+}

--- a/lib/main.js
+++ b/lib/main.js
@@ -59,7 +59,18 @@ event_sources.json and ${program.eventFile} files as needed.`)
 
     this._createSampleFile(program.eventFile, 'event.json')
     const splitHandler = program.handler.split('.')
-    const filename = splitHandler[0] + '.js'
+    const filename = (() => {
+      for (const extension of ['.js', '.mjs']) {
+        if (fs.existsSync(splitHandler[0] + extension)) {
+          return splitHandler[0] + extension
+        }
+      }
+    })()
+    if (filename == null) {
+      console.error('Handler file not found.')
+      process.exitCode = 255
+      return
+    }
     const handlername = splitHandler[1]
 
     // Set custom environment variables if program.configFile is defined

--- a/test/node-lambda.js
+++ b/test/node-lambda.js
@@ -189,7 +189,7 @@ describe('bin/node-lambda', () => {
       })
     })
 
-    describe('node-lambda run (Multiple events))', () => {
+    describe('node-lambda run (Multiple events)', () => {
       const eventObj = [{
         asyncTest: false,
         callbackWaitsForEmptyEventLoop: true,
@@ -217,7 +217,7 @@ describe('bin/node-lambda', () => {
       })
     })
 
-    describe('node-lambda run (disable Multiple events))', () => {
+    describe('node-lambda run (disable Multiple events)', () => {
       const eventObj = [{
         asyncTest: false,
         callbackWaitsForEmptyEventLoop: true,
@@ -264,7 +264,7 @@ describe('bin/node-lambda', () => {
       })
     })
 
-    describe('node-lambda run (API Gateway events))', () => {
+    describe('node-lambda run (API Gateway events)', () => {
       const eventObj = {
         asyncTest: false,
         callbackWaitsForEmptyEventLoop: true,

--- a/test/node-lambda.js
+++ b/test/node-lambda.js
@@ -295,6 +295,54 @@ describe('bin/node-lambda', () => {
         })
       })
     })
+
+    describe('node-lambda run (by *.mjs)', () => {
+      it('`node-lambda run` by index.mjs', function (done) {
+        const run = spawn('node', [
+          nodeLambdaPath, 'run',
+          '--handler', 'index_mjs.handler'
+        ])
+        let stdoutString = ''
+        run.stdout.on('data', (data) => {
+          stdoutString += data.toString().replace(/\r|\n|\s/g, '')
+        })
+
+        run.on('exit', (code) => {
+          const expected = 'Runningindex.handler(mjs)' +
+            '==================================' +
+            'event{asyncTest:false,callbackWaitsForEmptyEventLoop:true,callbackCode:\'callback(null);\'}' +
+            '==================================' +
+            'Stoppingindex.handler(mjs)Success:'
+          assert.equal(stdoutString, expected)
+          assert.equal(code, 0)
+          done()
+        })
+      })
+    })
+
+    describe('node-lambda run (handler file not found)', () => {
+      it('`node-lambda run` Invalid handler specification.', function (done) {
+        const run = spawn('node', [
+          nodeLambdaPath, 'run',
+          '--handler', 'not_found.handler'
+        ])
+        let stdoutString = ''
+        run.stdout.on('data', (data) => {
+          stdoutString += data.toString().replace(/\r|\n|\s/g, '')
+        })
+        let stderrString = ''
+        run.stderr.on('data', (data) => {
+          stderrString += data.toString()
+        })
+
+        run.on('exit', (code) => {
+          assert.equal(stdoutString, '')
+          assert.match(stderrString, /Handler file not found\./)
+          assert.equal(code, 255)
+          done()
+        })
+      })
+    })
   })
 
   describe('node-lambda duplicate check of short option', () => {

